### PR TITLE
Ingress rule configurability

### PR DESCRIPTION
**Pull request by mattl-netflix**
Make it possible to:
1. skip deleting ingress rules
2. skip updating ingress rules
3. only add ingress rules for public IPs

### CODE REVIEW

File: priam/src/main/java/com/netflix/priam/config/IConfiguration.java [hunk 1]
```diff
@@ -1111,6 +1111,26 @@ default BackupsToCompress getBackupsToCompress() {
...
+    /**
+     * @return true if Priam should skip updating ingress rules for ips found in the token database.
+     */
+    default boolean skipUpdatingOthersIngressRules() {
+        return false;
+    }
+
+    /** @return get the threshold at which point we might risk not getting our ingress rule set. */
+    default int getACLSizeWarnThreshold() {
+        return 500;
```
Comment (sumanth-pasupuleti) [inline priam/src/main/java/com/netflix/priam/config/IConfiguration.java]: Should we have a slightly lower threshold like 400 or 450 (given 500 is the known aws limit)

File: priam/src/main/java/com/netflix/priam/aws/UpdateSecuritySettings.java [hunk 2]
```diff
@@ -82,21 +77,25 @@ public void execute() {
...
-        if (!aclToRemove.isEmpty()) {
-            membership.removeACL(aclToRemove, port, port);
-            firstTimeUpdated = true;
+        if (!config.skipUpdatingOthersIngressRules()) {
+            Set<String> aclToAdd = Sets.difference(desiredAcl, currentAcl);
+            logger.info("ingress rules to update: {}", Joiner.on(",").join(aclToAdd));
+            if (!aclToAdd.isEmpty()) {
+                int resultingSize = aclToAdd.size() + currentAcl.size();
+                if (resultingSize > config.getACLSizeWarnThreshold()) {
+                    logger.warn("We are trying to make too many ingress rules! {}", resultingSize);
```
Comment (sumanth-pasupuleti) [inline priam/src/main/java/com/netflix/priam/aws/UpdateSecuritySettings.java]: Should we also start emitting a metric for resultingSize (in addition to log line). Wondering how discoverable a log line alone would be.

File: priam/src/main/java/com/netflix/priam/aws/AWSIPConverter.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import javax.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AWSIPConverter implements IPConverter {
```
Comment (sumanth-pasupuleti) [inline priam/src/main/java/com/netflix/priam/aws/AWSIPConverter.java:16]: nit: good to add comments to public classes

**Review by sumanth-pasupuleti — APPROVED** [review]

File: priam/src/main/java/com/netflix/priam/aws/AWSIPConverter.java (hunk 3 above)
Comment (mattl-netflix) [reply priam/src/main/java/com/netflix/priam/aws/AWSIPConverter.java:16]: Done.

File: priam/src/main/java/com/netflix/priam/config/IConfiguration.java (hunk 1 above)
Comment (mattl-netflix) [reply priam/src/main/java/com/netflix/priam/config/IConfiguration.java]: I removed this altogether, began publishing warnings when we (possibly) fail to update ingress rules, and began publishing a gauge of the size of the list every time it is measured.

File: priam/src/main/java/com/netflix/priam/aws/UpdateSecuritySettings.java (hunk 2 above)
Comment (mattl-netflix) [reply priam/src/main/java/com/netflix/priam/aws/UpdateSecuritySettings.java]: Point taken. Done.

**Pull request merged**